From 095a4f17e7a0aae76476150ec332bba6ecfd16bf Mon Sep 17 00:00:00 2001
From: c4pp4
Date: Wed, 21 Oct 2026 03:15:55 +0200
Subject: [PATCH 1/1] Do not warn about missing PulseAudio D-Bus interface

PipeWire does not provide the org.PulseAudio1 D-Bus interface, so the
warning is printed on every start. The interface is only needed for the
optional Stream Restore support, volume control works without it.

Signed-off-by: c4pp4
---
 src/volume-control-pulse.vala | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/volume-control-pulse.vala b/src/volume-control-pulse.vala
index 08955d0..ba21e24 100644
--- a/src/volume-control-pulse.vala
+++ b/src/volume-control-pulse.vala
@@ -738,7 +738,11 @@ public class VolumeControlPulse : VolumeControl
 				props_variant.get ("(v)", out props);
 				address = props.get_string ();
 			} catch (GLib.Error e) {
-				warning ("unable to get pulse unix socket: %s", e.message);
+				/* Not provided by PipeWire, which has no PulseAudio D-Bus interface */
+				if (e is DBusError.SERVICE_UNKNOWN)
+					debug ("unable to get pulse unix socket: %s", e.message);
+				else
+					warning ("unable to get pulse unix socket: %s", e.message);
 				return null;
 			}
 		}
-- 
2.55.0

